components/ord-service/src/main/java/com/sap/cloud/cmp/ord/service/controller/ODataController.java

```diff
@@ -1,30 +1,54 @@
 package com.sap.cloud.cmp.ord.service.controller;
 
+import java.io.IOException;
+import java.util.UUID;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.sap.olingo.jpa.processor.core.api.JPAClaimsPair;
 import com.sap.olingo.jpa.processor.core.api.JPAODataCRUDContextAccess;
+import com.sap.olingo.jpa.processor.core.api.JPAODataClaimsProvider;
 import com.sap.olingo.jpa.processor.core.api.JPAODataGetHandler;
+
 import org.apache.olingo.commons.api.ex.ODataException;
 import org.apache.olingo.server.api.debug.DefaultDebugSupport;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RequestMethod;
 
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
 
 @Controller
 @RequestMapping("/${odata.jpa.request_mapping_path}/**")
-public class ODataController {
+public class ODataController extends com.sap.cloud.cmp.ord.service.controller.Controller {
 
     @Autowired
     private JPAODataCRUDContextAccess serviceContext;
 
-    @RequestMapping(value = "**", method = { RequestMethod.GET })
-    public void handleODataRequest(HttpServletRequest request, HttpServletResponse response) throws ODataException {
+    private static final Logger logger = LoggerFactory.getLogger(ODataController.class);
+
+    @RequestMapping(value = "**", method = {RequestMethod.GET})
```

Comment:
`throws IOException` seems unneeded

Reply:
It comes from the usage of the `ObjectMapper` in `extractInternalTenantIdFromIDToken` method which is used in the Controller via the `createClaims` method.

Or do you suggest that we better catch it somewhere along the chain?

Reply:
No, it is fine